### PR TITLE
AirfRANS: T_max=10 — lr=7e-4 (between baseline and 1e-3)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

The phase transition mechanism at T_max=10 / epoch 40 is our primary lever for improving val_primary/surface_mse. The transition depth is stochastic and appears LR-sensitive:
- lr=5e-4 (baseline): 0.0207 best observed
- lr=3e-4 (gilbert, in-flight): unknown
- lr=1e-3 (senku, in-flight): unknown

lr=7e-4 fills the critical gap between baseline and the 1e-3 run. A slightly higher LR may push the model into a deeper basin at each cosine trough by providing more aggressive gradient steps before the LR drops to zero. This is a fine-grained sweep of the most sensitive hyperparameter for this phase-transition mechanism.

## Instructions

Run the following command:

```bash
cd target/icml2026 && python train.py \
  --dataset airfrans \
  --airfrans_task full \
  --optimizer adamw \
  --lr 7e-4 \
  --cosine_t_max 10 \
  --no-use-ema \
  --enable-fourier \
  --epochs 999 \
  --wandb_group airfrans-tmax10-lr-sweep
```

Key change vs baseline:
- `--lr 7e-4` (up from 5e-4, between baseline and the lr=1e-3 senku run)
- All other flags identical to baseline T_max=10 run (3L/192d, AdamW, Fourier)

**Critical:** The phase transition typically occurs around epoch 40 (cosine trough of cycle 4). Monitor val_primary/surface_mse carefully. Report:
1. Best val_primary/surface_mse and at which epoch it occurred
2. Whether the phase transition occurred and at what epoch
3. Training stability (any divergence or spikes?)

## Baseline

- **Dataset:** AirfRANS (full task)
- **Primary metric:** `val_primary/surface_mse` (lower is better)
- **Current best:** 0.0207 (val) — PR #2617 (T_max=10, lr=5e-4, 3L/192d, AdamW, epoch 40)
- **Previous:** 0.0248 (PR #2556) — same config, different seed, phase transition is stochastic
- **External target:** 0.0043 (~4.8x gap remaining)
- **Baseline W&B run:** z7t3ibwi
- **In-flight LR experiments:** lr=3e-4 (gilbert #2614), lr=1e-3 (senku #2615)
- **Reproduce baseline:** `cd target/icml2026 && python train.py --dataset airfrans --airfrans_task full --optimizer adamw --lr 5e-4 --cosine_t_max 10 --no-use-ema --enable-fourier --epochs 999`

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT emma:

## Results

### AirfRANS lr=7e-4 with T_max=10 cosine schedule

| Metric | Baseline (lr=5e-4) | This run (lr=7e-4) | Delta |
|--------|-------------------|--------------------|----|
| **val_primary/surface_mse** | 0.0207 (epoch 40) | **0.0184** (epoch 35) | **-11.1%** ✅ |
| val surface_mse_p | — | 0.0735 | — |
| val volume_mse | — | 0.133 | — |
| test_primary/surface_mse | TBD | 0.2323 (at final epoch 41, cosine peak — not representative) | — |

**Best val_primary/surface_mse: 0.0184 at epoch 35 — beats baseline 0.0207 by 11.1%**

### Phase transition analysis

The phase transition arrived **5 epochs earlier** (epoch 35) with lr=7e-4 compared to the baseline's epoch 40 with lr=5e-4. The higher LR accelerates the staircase descent into deeper basins at each cosine trough.

Trough envelope progression (val_primary/surface_mse at cosine troughs):
- Epoch 10: 0.230
- Epoch 20: 0.076
- Epoch 24: 0.034
- Epoch 30: 0.031
- **Epoch 35: 0.018** (best — phase transition)
- Epoch 40: 0.031 (did not beat epoch 35)

### Training stability

The higher LR causes significantly **more volatile oscillations** between cosine troughs:
- Peak-to-trough swings: epochs 38 and 41 spiked to 0.264-0.266 (surface_mse_p > 1.0)
- The epoch 38 spike (0.264) disrupted the epoch 40 trough, preventing it from beating epoch 35
- The test metrics (0.2323) were evaluated at epoch 41, which was on a cosine peak — **not representative of the best model**

### What happened

**The hypothesis is confirmed** — lr=7e-4 pushes the model into a deeper basin earlier than lr=5e-4. The phase transition occurred at epoch 35 instead of epoch 40, and reached a deeper minimum (0.0184 vs 0.0207). However, the higher LR also introduces much larger oscillations at cosine peaks, which can destabilize convergence at later troughs (epoch 40 couldn't match epoch 35).

This suggests lr=7e-4 is close to an optimal LR for this phase-transition mechanism — aggressive enough to accelerate the transition but volatile enough that later troughs may not consistently improve.

### Suggested follow-ups

1. **lr=6e-4** — may offer the acceleration of 7e-4 with less volatility, potentially allowing epoch 40+ troughs to also improve
2. **Longer training with lr=7e-4** — the epoch 40 trough was disrupted by the epoch 38 spike. With more epochs, later troughs (50, 60) might recover and push deeper
3. **LR warmup + lr=7e-4** — a warmup period could reduce the early oscillation amplitude while preserving the deeper basin effect at troughs
4. **Evaluate test metrics at best-val checkpoint** — the current test eval at the final epoch is misleading. Saving and evaluating the best-val checkpoint would give a more accurate test metric

### Experiment details

- **Command:** `cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 python train.py --dataset airfrans --airfrans-task full --optimizer adamw --lr 7e-4 --cosine-t-max 10 --no-use-ema --enable-fourier --epochs 999 --agent emma --wandb-name "emma/airfrans-tmax10-lr7e4" --wandb-group airfrans-tmax10-lr-sweep`
- **W&B run:** 3pbxocca
- **Epochs:** 41 (30-min timeout)
- **GPU:** NVIDIA RTX PRO 6000 Blackwell Server Edition (98GB)

**Comment by tcapelle** [comment]
ADVISOR: Merging — 0.01841 beats the AirfRANS baseline of 0.0197 by 6.5%. lr=7e-4 produces an earlier and deeper phase transition (epoch 35 vs epoch 38-40 for lower LRs). This is now the new AirfRANS baseline. Excellent discovery, emma!
